src/bindings.cpp

```diff
@@ -157,9 +157,17 @@ void deinit()
         delete cache;
         cache = nullptr;
     }
+
     token_deinit();
 }
 
+void __attribute__((destructor)) fini(void)
+{
+    deinit();
+    GDALDestroy();
+    usleep(680000); // nearly an eternity
```

Comment:
I'm surprised this doesn't block a thread from termination...

Reply:
This routine only executes when the shared library is unloaded (in context of Java, only when the static `GDALWarp` class is finalized(?)), so typically only at process termination.  The `sleep` may actually not be required.